docs/src/main/paradox/file.md

```diff
@@ -68,6 +68,29 @@ Scala
 Java
 : @@snip (../../../../file/src/test/java/akka/stream/alpakka/file/javadsl/DirectoryChangesSourceTest.java) { #minimal-sample }
 
+### LogRotationSink
+
+The `LogRotationSink` will create and write to multiple files.  
+This sink will wait for a function as parameter which returns a
+ `Bytestring => Option[Path]` function. If the generated function returns a path
+ the sink will rotate the file output to this new path and the actual `ByteString` will be
+  written to this new file too.
+ With this approach the user can define a custom stateful file generation implementation.
+
+A small snippet for the usage:
+
+Scala
+: @@snip (../../../../file/src/test/scala/akka/stream/alpakka/file/scaladsl/LogRotatorSinkSpec.scala) { #LogRotationSink-sample }
+
+In this sample we create a size based rotation function:
+
+Scala
+: @@snip (../../../../file/src/test/scala/akka/stream/alpakka/file/scaladsl/LogRotatorSinkSpec.scala) { #LogRotationSink-filesize-sample }
+
+In this sample we create a time based rotation function:
+
+Scala
+: @@snip (../../../../file/src/test/scala/akka/stream/alpakka/file/scaladsl/LogRotatorSinkSpec.scala) { #LogRotationSink-timebased-sample }
```

Comment:
We need the corresponding Java factory and Java samples as well, should be piece of cake with everything else in place! 🍰